### ISSUE TITLE
Delete.php

### ISSUE THREAD

**Issue by kavfam** [taniarascia/mysql#11]
$id = $_POST["submit"];

Is that line correct?

Should it be 

$id = $_GET["id"];

**Comment by ronraney** [comment]
Yes - I was going to ask about the same thing.
In the tutorial, it's written as this:
```
if (isset($_GET["id"])) {
  //if (!hash_equals($_SESSION['csrf'], $_POST['csrf'])) die();
  try {
    $pdo = new PDO($dsn, $DBuser, $DBpass, $options);

    $id = $_GET["id"];
```
I have an additional issue. I cannot get the CSRF to work on the delete page. I get this error when I add the code that is commented above:
_Warning: hash_equals(): Expected user_string to be a string, null given in /var/www/html/crud-app/delete.php on line 13_

At the end of the tutorial, it says to add this to all of the /public pages:
if (isset($_POST['submit'])) {
  if (!hash_equals($_SESSION['csrf'], $_POST['csrf'])) die();

When I do this, clicking the Delete button does nothing. I'm not sure if it should be $_GET or $_POST, and not sure why the 'csrf' is working on other pages, but not this one.

**Comment by djhayman** [comment]
As per the HTTP specification (relevant section: https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html), GET and HEAD requests should not have side effects, such as deletion of a resource. It is correct to use POST here (or even better, DELETE if it was an API).

The page itself is set up to use a form with POST method, and the submit button's value is used to specify the ID.

**Comment by djhayman** [comment]
OK - so some of the confusion here looks like it's because Tania hasn't updated her blog tutorial to match all the changes in this repo.

**Comment by ronraney** [comment]
In my case, the missing code was this:
`<button type="submit" name="submit" value="<?php echo escape($row["id"]); ?>">Delete</button></td>`

Originally, I hadn't changed from $_GET to $_POST. Once I did this, it still didn't work. I realized that my "submit" was incorrect, so I changed to the above. It now works. Thanks for the help.

**Comment by taniarascia** [comment]
Hiya, sorry, I haven't looked at this repo for quite a while. In that last comment, it seems to me like you didn't change it to match the code in the repo, but it works based on the code in the repo...I'm a bit confused. Can you let me know what error exists either in the repo or the tutorial and I can update it?

**Comment by ronraney** [comment]
Hello. It's my experience that the code given in the tutorial doesn't match what's in Github. There may have been a disconnect on my end, making the additional CSRF code changes. I had the same problem as the original poster of this issue, which was $_GET vs $_POST. This confusion came from what is shared in the tutorial vs. what's in Github. 

I piggybacked on the issue, further complicating it a bit. I changed the method to $_POST, but it still didn't work. Later on, I realized my "submit" markup was wrong. I changed it to what's in Github, and it works now. My issue is resolved by changing to what's in Github.I have a different system in place using Docker, and I can't simply copy/paste from Github.

I will edit my previous comment to be more clear.
